src/components/radio-group/daikin-radio-group.ts

```diff
@@ -0,0 +1,221 @@
+ * - `daikin-input-group` > `daikin-radio-group` > `daikin-radio`
+ *
+ * @fires change - A custom event emitted when current checked radio changed.
+ *
+ * @slot - A slot for radio buttons. Place `daikin-radio` elements here.
+ *
+ * @example
+ *
+ * ```html
+ * <daikin-radio-group>
+ *   <daikin-radio name="name" value="value1" label="Option1"></daikin-radio>
+ *   <daikin-radio name="name" value="value2" label="Option2"></daikin-radio>
+ *   <daikin-radio name="name" value="value3" label="Option3"></daikin-radio>
+ * </daikin-radio-group>
+ * ```
+ *
+ * ```html
+ * <daikin-input-group label="Label text" helper="Helper text">
+ *   <daikin-radio-group>
+ *     <daikin-radio name="name" value="value1" label="Option1"></daikin-radio>
+ *     <daikin-radio name="name" value="value2" label="Option2"></daikin-radio>
+ *     <daikin-radio name="name" value="value3" label="Option3"></daikin-radio>
+ *   </daikin-radio-group>
+ * </daikin-input-group>
+ * ```
+ */
+@customElement("daikin-radio-group")
+export class DaikinRadioGroup extends LitElement {
+  static override readonly styles = css`
+    ${unsafeCSS(tailwindStyles)}
+
+    :host {
+      display: inline-block;
+    }
+  `;
+
+  @queryAssignedElements({ selector: "daikin-radio" })
+  private readonly _radios!: Array<DaikinRadio>;
+
+  /**
+   * Specify the radio group orientation
+   */
+  @property({ type: String })
+  orientation: RadioGroupProps["orientation"] = "horizontal";
+
+  /**
+   * The form name.
+   */
+  @property({ type: String, reflect: true })
+  name = "";
+
+  /**
+   * `value` of the currently selected radio.
+   * see {@link DaikinRadio.value}
+   */
+  @property({ type: String, reflect: true })
+  value = "";
+
+  private _updateRadioIndex(daikinRadio: DaikinRadio, index: string) {
+    const input = daikinRadio.shadowRoot?.querySelector("input");
+    input?.setAttribute("tabindex", index);
+  }
+
+  private _updateRadios(value: string, name: string) {
+    const defaultValue = this._radios.find(
+      (radio) => radio.value === this.value
+    );
+    for (const [index, daikinRadio] of this._radios.entries()) {
+      daikinRadio.name = name;
+      if (!defaultValue && index === 0) {
+        this._updateRadioIndex(daikinRadio, "0");
+        continue;
+      }
+      if (!defaultValue) {
+        this._updateRadioIndex(daikinRadio, "-1");
+        continue;
+      }
+      if (daikinRadio.value === value) {
+        this.value = value;
+        daikinRadio.checked = true;
+        this._updateRadioIndex(daikinRadio, "0");
+        daikinRadio.internals.setFormValue(daikinRadio.value);
+      } else {
+        daikinRadio.checked = false;
+        this._updateRadioIndex(daikinRadio, "-1");
+        daikinRadio.internals.setFormValue(null);
+      }
+    }
+  }
+
+  private readonly _handleRadioChange = (
+    event: CustomEvent<{ value: string }>
+  ) => {
+    this._updateRadios((event.target as HTMLInputElement).value, this.name);
```

Comment:
```suggestion
    this.value = (event.target as HTMLInputElement).value;
```

`_updateRadios()` will be done by `updated()` if we set `this.value`.